pkg/services/ngalert/store/alert_rule.go

```diff
@@ -275,9 +275,26 @@ func (st DBstore) ListAlertRules(ctx context.Context, query *ngmodels.ListAlertR
 		q = q.Asc("namespace_uid", "rule_group", "rule_group_idx", "id")
 
 		alertRules := make([]*ngmodels.AlertRule, 0)
-		if err := q.Find(&alertRules); err != nil {
+		var faultyRules int64 = 0
+		rule := new(ngmodels.AlertRule)
+		rows, err := q.Rows(rule)
+		if err != nil {
 			return err
 		}
+		defer rows.Close()
+		for rows.Next() {
```

Comment:
Could we add a comment explaining why we're doing this by hand over `.Find(&someSliceOfRules)`? Just to prevent someone from refactoring this and undoing it in the future
```suggestion
		// Deserialize each rule separately in case any of them contain invalid JSON.
		for rows.Next() {
```